test/fw/ptl/utils/plugins/ptl_test_runner.py

```diff
@@ -744,10 +749,300 @@ def __are_requirements_matching(param_dic=None, test=None):
+                self.hardware_report_timer.cancel()
+                raise PbsHardwareError(msg=_msg+"unable to get disk info")
+            elif 70 < used_disk_percent < 95:
+                _msg = hostname + ": used disk is "
+                self.logger.warning(_msg + str(used_disk_percent) + "%")
+            elif used_disk_percent > 95:
+                _msg = hostname + ": used disk is "
+                self.logger.error(_msg + str(used_disk_percent) + "%")
+                self.hardware_report_timer.cancel()
+                raise PbsHardwareError(msg=_msg+str(used_disk_percent)+"%")
+            # checks core files in mom_priv
+            pbs_conf = du.parse_pbs_config(hostname)
+            mom_priv_path = os.path.join(pbs_conf["PBS_HOME"],
+                                         "mom_priv")
+            mom_priv_files = du.listdir(hostname,
+                                        mom_priv_path, sudo=True,
+                                        fullpath=False)
+            for filename in mom_priv_files:
+                if filename.startswith("core"):
+                    _msg = hostname + ": core file exist at "
+                    self.logger.warning(_msg + mom_priv_path)
+                    self.hardware_report_timer.cancel()
+                    raise PbsHardwareError(msg=_msg + mom_priv_path)
+            # checks core files in user home directory
+            ptl_users = ['pbsuser1', 'pbsuser2', 'pbsuser3', 'pbsuser4',
+                         'pbsuser5', 'pbsuser6', 'pbsuser7', 'pbstest',
+                         'pbsroot', 'pbsadmin']
+            for user in ptl_users:
+                cmd = "getent passwd " + user
+                user_home = du.run_cmd(hostname, cmd=cmd)
+                if user_home['rc'] != 0:
+                    _msg = hostname + ", user-" + user
+                    self.logger.error(_msg + ": unable to get user home")
+                    self.hardware_report_timer.cancel()
+                    raise PbsHardwareError(
+                        msg=_msg+": unable to get user home")
+                else:
+                    user_home_list = user_home['out'][0].split(":")
+                    user_home_files = du.listdir(hostname,
+                                                 user_home_list[5],
+                                                 sudo=True,
+                                                 fullpath=False)
+                    for filename in user_home_files:
+                        if filename.startswith("core"):
+                            _msg = hostname
+                            _msg = ", user-" + user
+                            _msg += ": core file exist at "
+                            self.logger.warning(_msg + user_home['out'])
+                            self.hardware_report_timer.cancel()
+                            raise PbsHardwareError(msg=_msg+user_home['out'])
+        # server hardware report
+        for hostname in self.param_dict['servers']:
```

Comment:
there is a lot of repetition of code here, shall we combine the two for loops?

Reply:
i have used for loop in  new changes

Reply:
@prakashcv13 i have reviewed your all comments. if any suggestions, please suggest me.